R/rescale_weights.R

```diff
@@ -2,96 +2,159 @@
+#'
 #'   - Carle A.C. (2009). Fitting multilevel models in complex survey data
 #'   with design weights: Recommendations. BMC Medical Research Methodology
 #'   9(49): 1-13
 #'
-#'   - Asparouhov T. (2006). General Multi-Level Modeling with Sampling
-#'   Weights. Communications in Statistics - Theory and Methods 35: 439-460
+#'   - Kish, L. (1965) Survey Sampling. London: Wiley.
+#'
+#' @examplesIf all(insight::check_if_installed(c("lme4", "parameters"), quietly = TRUE))
+#' data(nhanes_sample)
+#' head(rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR"))
+#'
+#' # also works with multiple group-variables
+#' head(rescale_weights(nhanes_sample, c("SDMVSTRA", "SDMVPSU"), "WTINT2YR"))
+#'
+#' # or nested structures.
+#' x <- rescale_weights(
+#'   data = nhanes_sample,
+#'   by = c("SDMVSTRA", "SDMVPSU"),
+#'   probability_weights = "WTINT2YR",
+#'   nest = TRUE
+#' )
+#' head(x)
+#'
+#' \donttest{
+#' # compare different methods, using multilevel-Poisson regression
 #'
-#' @examples
-#' if (require("lme4")) {
-#'   data(nhanes_sample)
-#'   head(rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR"))
-#'
-#'   # also works with multiple group-variables
-#'   head(rescale_weights(nhanes_sample, c("SDMVSTRA", "SDMVPSU"), "WTINT2YR"))
-#'
-#'   # or nested structures.
-#'   x <- rescale_weights(
-#'     data = nhanes_sample,
-#'     by = c("SDMVSTRA", "SDMVPSU"),
-#'     probability_weights = "WTINT2YR",
-#'     nest = TRUE
-#'   )
-#'   head(x)
-#'
-#'   nhanes_sample <- rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR")
-#'
-#'   glmer(
-#'     total ~ factor(RIAGENDR) * (log(age) + factor(RIDRETH1)) + (1 | SDMVPSU),
-#'     family = poisson(),
-#'     data = nhanes_sample,
-#'     weights = pweights_a
-#'   )
+#' d <- rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR")
+#' result1 <- lme4::glmer(
+#'   total ~ factor(RIAGENDR) + log(age) + factor(RIDRETH1) + (1 | SDMVPSU),
+#'   family = poisson(),
+#'   data = d,
+#'   weights = rescaled_weights_a
+#' )
+#' result2 <- lme4::glmer(
+#'   total ~ factor(RIAGENDR) + log(age) + factor(RIDRETH1) + (1 | SDMVPSU),
+#'   family = poisson(),
+#'   data = d,
+#'   weights = rescaled_weights_b
+#' )
+#'
+#' d <- rescale_weights(
+#'   nhanes_sample,
+#'   probability_weights = "WTINT2YR",
+#'   method = "kish"
+#' )
+#' result3 <- lme4::glmer(
+#'   total ~ factor(RIAGENDR) + log(age) + factor(RIDRETH1) + (1 | SDMVPSU),
+#'   family = poisson(),
+#'   data = d,
+#'   weights = rescaled_weights
+#' )
+#' parameters::compare_parameters(
+#'   list(result1, result2, result3),
+#'   exponentiate = TRUE,
+#'   column_names = c("Carle (A)", "Carle (B)", "Kish")
+#' )
 #' }
 #' @export
-rescale_weights <- function(data, by, probability_weights, nest = FALSE) {
+rescale_weights <- function(data,
+                            by = NULL,
+                            probability_weights = NULL,
+                            nest = FALSE,
+                            method = "carle") {
```

Comment:
Can we have a `match.arg()` to check values passed to `method`? Especially since the `switch()` call below means that passing `method = "carl"` would run the Kish method.